client/src/components/assessments/OndemandAssessments.js

```diff
@@ -0,0 +1,235 @@
+import { gql } from "@apollo/client"
+import API from "api"
+import AssessmentModal from "components/assessments/AssessmentModal"
+import ConfirmDestructive from "components/ConfirmDestructive"
+import { ReadonlyCustomFields } from "components/CustomFields"
+import Fieldset from "components/Fieldset"
+import LinkTo from "components/LinkTo"
+import Model, { NOTE_TYPE } from "components/Model"
+import PeriodsNavigation from "components/PeriodsNavigation"
+import { Formik } from "formik"
+import moment from "moment"
+import { PeriodsDetailsPropType } from "periodUtils"
+import PropTypes from "prop-types"
+import React, { useCallback, useEffect, useMemo, useState } from "react"
+import { Button, Panel, Table } from "react-bootstrap"
+import { toast } from "react-toastify"
+import REMOVE_ICON from "resources/delete.png"
+import Settings from "settings"
+
+const GQL_DELETE_NOTE = gql`
+  mutation($uuid: String!) {
+    deleteNote(uuid: $uuid)
+  }
+`
+
+const OnDemandAssessments = ({
+  entity,
+  entityType,
+  subEntities,
+  style,
+  periodsDetails,
+  canAddAssessment,
+  onUpdateAssessment
+}) => {
+  /* recurrence has the value 'ondemand' for this specific assessment type and
```

Comment:
Should you check whether this is actually called with the value `ondemand` and not with anything else?

Reply:
For this specific case, I am aware that it does not really matter because it always gets the value _ondemand_. However, I wanted to grab the value as a prop to be consistent with the rest of the assessment implementations.

Reply:
What I meant is: add validation in the component, e.g. `if (recurrence !== RECURRENCE_TYPE.ON_DEMAND) { throw "some exception" }`